src/ddmd/mars.d

```diff
@@ -1316,6 +1318,8 @@ Language changes listed by -transition=id:
         /* At this point, name is the D source file name stripped of
          * its path and extension.
          */
+        if (strcmp(name, "-") == 0)
+            name = "__stdin";
```

Comment:
This is a hack to get around the fact that `-` is not a valid module name.  Also, this also gives us a saner basename for deriving the filenames of object files, ddoc output, etc..